### DISCUSSION TITLE
Does TCA support  x86_64-apple-ios-simulator?

### DISCUSSION THREAD

**Discussion by shshalom** [pointfreeco/swift-composable-architecture#1571] [Q&A] [answered]
### Description

I've been playing with TCA for a while now on a side project and everything worked great, and now  I've decided to bring it into our main app -

The app itself, like many other projects, is heavily dependent on 3rd party pods, which they are excluding arm64 for the simulator (['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64')

when I added TCA to the project I received the following error:
```swift
Could not find module 'ComposableArchitecture' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator, at: .../ComposableArchitecture.swiftmodule

This error goes away if I'll run xCode using Roessta - but this causes inconvenience to other team members.

Any ideas on how to solve this?


### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

To be able to run TCA on x86_64-apple-ios-simulator

### Actual behavior

when I added TCA to the project I received the following error:
```swift
Could not find module 'ComposableArchitecture' for target 'x86_64-apple-ios-simulator'; found: arm64-apple-ios-simulator, at: .../ComposableArchitecture.swiftmodule
```

### Steps to reproduce

1. Add TCA to any project that has a pod dependency that has the following: 

['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = 'arm64'

### The Composable Architecture version information

0.44.0

### Destination operating system

iOS 15

### Xcode version information

 13.4.1 (13F100)

### Swift Compiler version information

```shell
swift-driver version: 1.45.2 Apple Swift version 5.6.1 (swiftlang-5.6.0.323.66 clang-1316.0.20.12)
Target: x86_64-apple-macosx12.0
```

**Comment by stephencelis** [comment]
@shshalom I don't think the library does anything peculiar with architecture (arm64, etc.), so this seems like more of a CocoaPods question than a bug in our library. Since we use issues for bug tracking, I'm going to convert this to a discussion where folks with CocoaPods experience may be able to help.

**Comment by jshier** [comment, marked as answer]
This happens when excluding archs and trying to use SPM through Xcode, as Xcode doesn't pass the excluded archs setting down to the package. I'm not sure there's any way to get this to work properly but others may have suggestions. I'd suggest getting rid of (or maybe updating) whatever dependency you're using that requires you to exclude arm64.

**Reply by shshalom** [reply]
Thanks @jshier, sadly we're using Kotlin Multiplatform  which force us to exclude arm64... which mean that as for now I will not be able to use TCA using Swift Package Manager. 🫤

And on the other hand I don't wanna add my own Pod support - especially with the recent changes and new modules introduced (like clock) .

I did tried to create an XCFramework for TCA - but I've found that this is not an easy task either.
@stephencelis any suggestions before let this go - and sadly dropping the idea of using TCA?!

**Reply by mbrandonw** [reply]
> And on the other hand I don't wanna add my own Pod support - especially with the recent changes and new modules introduced (like clock) .

While I'm not as well-versed in cocoapods as I used to be, it's my impression that it would be quite straightforward for someone to maintain podspecs for TCA and all of its dependencies. In fact, someone (👋 @ldstreet) [did just that](https://github.com/ldstreet/swift-composable-architecture-cocoapods), though it hasn't been kept up-to-date.

If someone wants to manage these podspecs and keep them up-to-date we would happily link to it from the readme so that others using cocoapods aren't left in the cold. We just don't feel like we can personally take on this burden since we do not actively use cocoapods, and so it is likely we will break it regularly.

> I did tried to create an XCFramework for TCA - but I've found that this is not an easy task either.

I believe others have been successful in building an xcframework of TCA, so you may be able to share what your issue is and see if someone can help.

**Reply by shshalom** [reply]
Hey @mbrandonw, Thanks for this.. actually you manage to ignite my will to try once more.
- and seems like I am on the right path.. the only thing I need to figure out is how to translate Dependencies module to PodSpec since it not an actual repo ..

any idea?

**Reply by shshalom** [reply]
Just wanted to update, I've managed to create  and add pod support to the latest TCA (0.44.1) thanks to @mbrandonw, @[ldstreet](https://github.com/ldstreet/swift-composable-architecture-cocoapods/commits?author=ldstreet)
ldstreet..

Now I can use this wonderful framework :-)

If anyone is interested here's my pod specs : https://github.com/shshalom/ComposableArchitecturePods
